lib/Sema/TypeCheckType.cpp

```diff
@@ -1778,6 +1778,13 @@ namespace {
       return diags.diagnose(std::forward<ArgTypes>(Args)...);
     }
 
+    template <typename... ArgTypes>
+    InFlightDiagnostic diagnose(TypeRepr *repr, ArgTypes &&... Args) const {
```

Comment:
One more nit - maybe it should be called `invalidateAndDiagnose` or `diagnoseAndInvalidate` or `diagnoseInvalidRepr` or just `diagnoseInvalid` for clarity?

Reply:
I like `diagnoseInvalid` since the first argument is a `TypeRepr`

Reply:
Sure, let's go with that!

Reply:
Done!